pkg/operators/ebpf/struct.go

```diff
@@ -0,0 +1,222 @@
+// Copyright 2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ebpfoperator
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/cilium/ebpf/btf"
+	"gopkg.in/yaml.v3"
+
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/btfhelpers"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/datasource"
+	metadatav1 "github.com/inspektor-gadget/inspektor-gadget/pkg/metadata/v1"
+)
+
+type Field struct {
+	metadatav1.Field
+	Tags   []string
+	Type   reflect.Type
+	Offset uint32
+	parent int
+}
+
+type Struct struct {
+	Fields []*Field `yaml:"fields"`
+	Size   uint32
+}
+
+func (f *Field) FieldName() string {
+	return f.Name
+}
+
+func (f *Field) FieldSize() uint32 {
+	return f.Size
+}
+
+func (f *Field) FieldOffset() uint32 {
+	return f.Offset
+}
+
+func (f *Field) FieldTags() []string {
+	return f.Tags
+}
```

Comment:
It's implementing the interfaces defined [here](https://github.com/inspektor-gadget/inspektor-gadget/pull/2512/commits/c277f4421911ddfcd2864860e61585aaff6930e4#diff-c47c1b934181a298f81910c970357de8c5adf806aeabf622eeae3e95fa275a86R39-R62) - this way DataSources doesn't need to know the actual structs behind it and it can stay private.